apps/web/src/components/Common/TopicSelector.tsx

```diff
@@ -0,0 +1,130 @@
+import { useFormatMessage } from "@litespace/ui/hooks/intl";
+import React, { useCallback, useMemo, useState } from "react";
+import TopicChanger from "@/components/Common/TopicChanger";
+import { useTopics, useUserTopics } from "@litespace/headless/topic";
+import { useUpdateUserTopics } from "@litespace/headless/user";
+import { TopicSelectionDialog } from "@litespace/ui/TopicSelectionDialog";
+import { useToast } from "@litespace/ui/Toast";
+import { useInvalidateQuery } from "@litespace/headless/query";
+import { QueryKey } from "@litespace/headless/constants";
+import { useUserContext } from "@litespace/headless/context/user";
+import { IUser } from "@litespace/types";
+
+export const TopicSelector: React.FC = () => {
+  const [showTopicsDialog, setShowTopoicsDialog] = useState<boolean>(false);
+  const { user } = useUserContext();
+
+  const intl = useFormatMessage();
+  const toast = useToast();
+  const invalidateQuery = useInvalidateQuery();
+
+  const isStudent = useMemo(() => user?.role === IUser.Role.Student, [user]);
+
+  const topics = useTopics({});
+  const userTopics = useUserTopics();
+
+  const allTopics = useMemo(() => {
+    if (!topics.query.data) return [];
+    return topics.query.data.list.map((topic) => ({
+      id: topic.id,
+      label: topic.name.ar,
+    }));
+  }, [topics.query.data]);
+
+  const selectedUserTopics = useMemo(() => {
+    if (!userTopics.data) return [];
+    return userTopics.data.map((topic) => ({
+      id: topic.id,
+      label: topic.name.ar,
+    }));
+  }, [userTopics]);
+
+  const userTopicsIds = useMemo(() => {
+    if (!userTopics.data) return [];
+    return userTopics.data.map((topic) => topic.id);
+  }, [userTopics]);
+
+  const onTopicChangeSuccess = useCallback(() => {
+    invalidateQuery([QueryKey.FindUserTopics]);
+    setShowTopoicsDialog(false);
+  }, [invalidateQuery]);
+
+  const onTopicChangeError = useCallback(() => {
```

Comment:
- It is better to name it `onError`
- Remember to report the error to Sentry using `capture`

```suggestion
  const onTopicChangeError = useCallback(() => {
```